### PR TITLE
Don't use fallback for missed blocks

### PR DESCRIPTION
**Pull request by vgorkavenko**
Now we will not try to re-fetch block info if it is missed

### CODE REVIEW

File: src/modules/lighthouse/Eth2Client.ts [hunk 1]
```diff
@@ -304,17 +320,19 @@ export class Eth2Client {
...
+        ).catch(
           (e) => {
             this.logger.error('Unexpected status code while fetching proposer duties info');
             throw e;
           }
         )
       );
       if (res.dependent_root != dependentRoot) {
-        throw new BeaconChainGeneralApiError(errCommon(
+        throw Error(
```
Comment (dgusakov) [inline src/modules/lighthouse/Eth2Client.ts:334]: Why you switch to general error here?

Comment (vgorkavenko) [reply src/modules/lighthouse/Eth2Client.ts:334]: `BeaconChainGeneralApiError` should only be used in basic requests functions

File: src/modules/lighthouse/Eth2Client.ts [hunk 2]
```diff
@@ -331,73 +349,41 @@ export class Eth2Client {
...
       .catch(rejectDelay(this.environment.ETH2_BEACON_RPC_RETRY_DELAY_MS))
-      .catch(() => retry(() => postFunc(subUrl, params)))
-      .catch((e) => {
-        this.logger.error('Error while doing ETH2 RPC request. Will try to switch to another RPC');
-        this.switchToNextRpc();
+      .catch(() => retry(() => callback(this.rpcUrls.main)))
+      .catch((e: any) => {
+        if (fallbackConditionCallback(e)) {
+          this.logger.error('Error while doing ETH2 RPC request. Will try to switch to another RPC if exists');
+          return retry(() => callback(this.rpcUrls.backup || this.rpcUrls.main));
```
Comment (madlabman) [inline src/modules/lighthouse/Eth2Client.ts:365]: May be it is better to basically throw an error if not backup url provided?

Comment (vgorkavenko) [reply src/modules/lighthouse/Eth2Client.ts:365]: It seems to me, a good option would be to write a warning and not try to use backup url

Comment (vgorkavenko) [reply src/modules/lighthouse/Eth2Client.ts]: fixed

**Review by dgusakov — APPROVED** [review]

**Review by madlabman — APPROVED** [review]

**Pull request merged**